crypto/asn1/a_verify.c

```diff
@@ -21,6 +21,32 @@
 #include "crypto/asn1.h"
 #include "crypto/evp.h"
 
+EVP_MD_CTX *evp_md_ctx_new_ex(EVP_PKEY *pkey, const ASN1_OCTET_STRING *id,
+                              OPENSSL_CTX *libctx, const char *propq)
+{
+    EVP_MD_CTX *ctx;
+    EVP_PKEY_CTX *pctx = NULL;
+
+    if ((ctx = EVP_MD_CTX_new()) == NULL
+        || (pctx = EVP_PKEY_CTX_new_from_pkey(libctx, pkey, propq)) == NULL) {
+        ASN1err(0, ERR_R_MALLOC_FAILURE);
```

Comment:
I wonder if this function should be somewhere else.. It doesnt really do much with ASN1. It just gets data and a length.

Reply:
Good point - so I've moved the function from `crypto/asn1/a_verify.c` to `crypto/evp/digest.c`.